README.rst

```diff
@@ -0,0 +1,218 @@
+Mail BB Gun
+===========
+
+A baby version of MailGun
+
+Requirements
+------------
+
+* Python 3.6
+* Docker
+* SMTP Server account
+
+Setup
+-----
+
+Start docker services.
+
+.. code-block:: bash
+
+    docker-compose up -d
+
+Install Python dependencies.  Using a Virtual Environment is highly recommended.
+
+.. code-block:: bash
+
+   pip install -r requirements.txt
+
+Generate the database password.  Be sure to save it in a safe place.
+
+.. code-block:: bash
+
+    export PG_PASSWORD=$(dd if=/dev/urandom bs=1 count=24 2>/dev/null | hexdump -v -e '/1 "%02X"')
+    echo $PG_PASSWORD
+
+Export your SMTP password.
+
+.. code-block:: bash
+
+    export SMTP_PASSWORD=YOUR_SMTP_PASSWORD
+
+Copy the sample config file and update it with relevant SMTP account values for your deployment.
+
+.. code-block:: bash
+
+    cp config.py.sample config.py
+
+Add the mailbbgun user to postgresql.
```

Comment:
PostgreSQL does provide a few replication options out of the box.  The decision of which one(s) to use will depend on the HA goals of the system.  I think streaming replication would be a good starting point.  Unfortunately, PostgreSQL does not provide automatic failover, so promoting the standby node to be the new master still requires manual intervention.

If automatic failover was a requirement, then I would look into deploying pgpool-II.  It would also give us the option to load balance read queries to the hot standby nodes.